src/effects/VST/VSTEffect.cpp

```diff
@@ -1316,9 +1316,32 @@ bool VSTEffectInstance::RealtimeProcessStart(MessagePackage& package)
       const auto len = chunk.size();
       const auto data = chunk.data();
       callSetChunk(true, len, data, &info);
+
+      // Emulate what happens for plugins like DeeGain,
+      // i.e. after setting the chunk, the vst callback is called
+      // with message audioMasterAutomate for all parameters.
+      //
+      // This fixes #3842 (for BlueCat Chorus and possibly other plugins)
+      // and does not harm plugins like DeeGain either
+      //
+      for (size_t i = 0; i < message.mParamsVec.size(); i++)
+      {
+         if (message.mParamsVec[i])
+            Automate(i, *message.mParamsVec[i]);
+      }
+
       for (auto& slave : mSlaves)
+      {
          slave->callSetChunk(true, len, data, &info);
 
+         // Emulation of DeeGain-type plugins for the slaves too
+         for (size_t i = 0; i < message.mParamsVec.size(); i++)
+         {
+            if (message.mParamsVec[i])
+               slave->Automate(i, *message.mParamsVec[i]);
+         }
+      }
```

Comment:
Great idea!  I’m guessing however that it is enough to do this to self only, and not needed for slaves.  Can you confirm?  If I am right, then make the minimal needed change.

Reply:
Also be sure to test the changing of presets while playing and while not.

I haven't tried but I think this may fail to fix it during play, because VSTEffectInstance::Automate required a check that it runs on the main thread, and otherwise does nothing.  That was necessary for me to add, to avoid deadlocks.

So:  the final fix may require you to wrap the call in a lambda as argument to `BasicUI::CallAfter`.